confidential-transfer/proof-generation/src/withdraw.rs

```diff
@@ -1,3 +1,44 @@
+//! Generates the zero-knowledge proofs required for a confidential withdraw.
+//!
+//! A confidential withdraw operation converts a user's encrypted, confidential token balance back into a
+//! standard, publicly-visible SPL token balance. To ensure this operation is valid and that a user
+//! cannot create tokens out of thin air, it requires two distinct zero-knowledge proofs.
+//!
+//! ## Protocol Flow and Proof Components
+//!
+//! 1.  **Calculate Remaining Balance**: The client first calculates the remaining confidential balance
+//!     by subtracting the desired `withdraw_amount` from their current known balance.
+//!
+//! 2.  **Homomorphic Calculation**: The client homomorphically computes the new encrypted balance
+//!     ciphertext. This is done by taking the current `available_balance` ciphertext and subtracting
+//!     a newly-encoded ciphertext of the `withdraw_amount`.
+//!
+//! 3.  **Generate Proofs**: The user generates two proofs to certify the validity of the operation:
+//!
+//!     -   **Ciphertext-Commitment Equality Proof (`CiphertextCommitmentEqualityProofData`)**:
+//!         This proof is the cryptographic link between the on-chain state and the solvency check. It
+//!         certifies that the newly computed `remaining_balance_ciphertext` (which will be the new
+//!         on-chain state) and a new Pedersen commitment both hide the exact same numerical value.
+//!         This is essential because the range proof operates on Pedersen commitments, not directly on
+//!         ElGamal ciphertexts.
+//!
+//!     -   **Ciphertext-Commitment Equality Proof (`CiphertextCommitmentEqualityProofData`)**:
+//!         This proof provides a cryptographic link that enables the solvency check. When the
+//!         `remaining_balance_ciphertext` is computed homomorphically, the prover may not know the
+//!         corresponding Pedersen opening (randomness) for the resulting ciphertext. Performing a
+//!         range proof requires knowledge of this opening.
+//!
+//!         To solve this, the prover creates a *new* Pedersen commitment for the remaining balance,
+//!         for which it knows the opening. The equality proof then certifies that the
+//!         homomorphically-derived ciphertext and this new commitment hide the exact same numerical
+//!         value. This allows the range proof to be performed on the new commitment.
```

Comment:
Were these meant to be very similar?

Reply:
Oh oops, sorry. I removed the duplicate description of the equality proof.